lib/cosmos/gui/qt.rb

```diff
@@ -531,14 +539,12 @@ def selection_end_line
   end
 
   private
-
   def escape_text(text, color)
+    @@color_cache[color] ||= "#%02X%02X%02X" % [color.red, color.green, color.blue]
     # You might think gsub! would use less memory but benchmarking proves gsub
     # with a regular express argument is the fastest and uses the least memory.
     # However, this is still an expensive operation due to how many times it is called.
-    text = text.gsub(/&/,'&amp;'.freeze).gsub(/\n/,'<br/>'.freeze).gsub(/\s/, '&nbsp;'.freeze).gsub(/>/,'&gt;'.freeze).gsub(/</,'&lt;'.freeze)
-    @@color_cache[color] ||= "#%02X%02X%02X" % [color.red, color.green, color.blue]
-    "<font color=\"#{@@color_cache[color]}\">#{text}</font>"
+    "<font color=\"#{@@color_cache[color]}\">#{text.gsub(@@regex,@@mapping)}</font>"
```

Comment:
This substitutes all the variables in one go and thus reduces the calls to gsub. It saves us about 2.5%.